frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -244,14 +306,31 @@ export class Gallery extends React.Component {
     switch (fetchContentList.apiResponse.status) {
       case 200: {
         const images = fetchContentList.body.filter(c => c.content_type === 'file').map(c => ({ src: '', contentId: c.content_id }))
-
         const imagesPreviews = await this.loadPreview(images)
 
-        this.setState({ imagesPreviews, imagesPreviewsLoaded: true })
+        // If the list of pictures is modified and we already were displaying
+        // a picture, we attempt to set the currently displayed picture
+        // to this picture.
+        // We fall back to the index that was already here.
+
+        let displayedPictureIndex = state.displayedPictureIndex
+        const currentPictureId = this.currentPictureId()
+        if (currentPictureId) {
+          for (let i = 0; i < imagesPreviews.length; i++) {
+            if (imagesPreviews[i].contentId === currentPictureId) {
+              displayedPictureIndex = i
+              break
+            }
+          }
```

Comment:
using `findIndex` this would be
`displayedPictureIndex = imagesPreviews.findIndex(image => image.contentId === currentPictureId)`

Reply:
The equivalent code using findIndex is:

```suggestion
        let displayedPictureIndex = -1

        const currentPictureId = this.currentPictureId()
        if (currentPictureId) {
          displayedPictureIndex = imagesPreviews.findIndex(image => image.contentId === currentPictureId)
        }

        if (displayedPictureIndex === -1) {
          displayedPictureIndex = state.displayedPictureIndex
        }
```

I prefer the for version, but feel free to commit my suggestion if you really prefer this version.

Reply:
I'm not sure I understand, your suggestion's addition seems to do more than its deletion

The findIndex is only meant to replace the for, why do you need some more variable initialization using findIndex ?

Reply:
sorry, my snippet is for replacing lines 316 and 317 too:

```
        let displayedPictureIndex = state.displayedPictureIndex
        const currentPictureId = this.currentPictureId()
```
findIndex is nice but it needs a bit of work here because it can return -1 and I need to add a check for that. The for loop was easier and the whole code with the for loop is more concise. I you really prefer I'd use findIndex here I'll commit an equivalent code using findIndex but the improvement by doing that is not clear to me.

Reply:
What's the difference between "displayed"Picture and "current"Picture ?

Reply:
I do prefer the use of `findIndex` but I'd write it with a little difference:
``` jsx
let displayedPictureIndex = -1
const currentPictureId = this.currentPictureId()

if (currentPictureId) {
  displayedPictureIndex = imagesPreviews.findIndex(image => image.contentId === currentPictureId)
  if (displayedPictureIndex === -1) displayedPictureIndex = state.displayedPictureIndex
}
```
The 2 reasons for me to prefer this are:
- it explicitly tells what we are doing if `currentPictureId` isn't in `imagesPreviews`
- in Tracim, we tend to use array manipulation functions over for loop

Reply:
> use most recent javascript features

This is not my goal. My goal is to write efficient and legible code, and for loops often achieve both goals for me, the latter being more than subjective though. I'm fond of recent Javascript feature and will gladly use them when needed.
Functional-style array manipulation functions invite developers to write code that loops over arrays more times than necessary and I don't think this is a good thing too.

They also force the usage of callback functions that get called for every elements of arrays, and calling a function is slow against running the body of a for loop.

Comment taken though, I'll try to avoid for loops and use functional-style programming instead in Tracim, but I'm not sure it is a good solution in every cases.

I agree with you about being explicit though.

In any case, I updated the code to use findIndex :-)